src/components/ShowList.jsx

```diff
@@ -2,19 +2,32 @@ import React from 'react';
 import { useCollection } from 'react-firebase-hooks/firestore';
 import db from '../lib/firebase';
 import Navigation from './Navigation';
+import { useHistory } from 'react-router-dom';
+import './ShowList.css';
 
 function ShowList() {
   const token = localStorage.getItem('token');
+  const history = useHistory();
   const [value, loading, error] = useCollection(
     db.collection('items').where('token', '==', token),
   );
 
+  const handleClick = () => {
+    history.push('/add-view');
+  };
+
   return (
-    <div>
-      <h1>ListView</h1>
+    <div className="list-view">
+      <h1>Smart Shopping List</h1>
       {error && <p>Error</p>}
       {loading && <p>Loading..</p>}
-      {value && (
+      {!loading && value.docs.length === 0 && (
```

Comment:
Just out of interest, I was wondering about why `!loading` was added in line 24. Does something unexpected happen on the frontend when left out?

Reply:
Hi Claire, this is a really great question! We tried it without, because we assumed it would work, but it actually threw an error when we tried to access the properties of the 'value' object before it was loaded. By putting this in we got rid of the TypeErrors

Reply:
A nice solution is to use [Conditional rendering ](https://reactjs.org/docs/conditional-rendering.html) in React  like this: {loading ? <p>Loading..</p> :  value.docs.length === 0}

Reply:
I second @bsinoimeri 👍